src/sat.cc

```diff
@@ -126,122 +127,98 @@ bool Sat::CheckEnvironment() {
 
   // Check if the cpu frequency test is enabled and able to run.
   if (cpu_freq_test_) {
-    if (!CpuFreqThread::CanRun()) {
-      logprintf(0,
-                "Process Error: This platform does not support this "
-                "test.\n");
-      bad_status();
+    if (!CpuFreqThread::CanRun(setup_step)) {
       return false;
```

Comment:
Same here. Why there is no error / logging in this case?

Reply:
This time we handle it in in the inner class - this test throws multiple errors for everything. If I need to pass in a test step for other reasons (aka other logs), then I do so and handle the error at the lower level. If I don't need to pass it in, then I handle it in the outer layer. This original code jus throws two errors.